client/src/components/mapping/FhirMappingPanel/FhirResourceTree/index.tsx

```diff
@@ -60,6 +61,16 @@ const FhirResourceTree = ({ onClickCallback }: Props) => {
   const attributesForResource: { [k: string]: IAttribute } = useSelector(
     (state: IReduxStore) => state.resourceInputs.attributesMap
   );
+  const attributesWithInputs = Object.keys(attributesForResource).filter(
+    path => attributesForResource[path].inputGroups.length > 0
+  );
+
+  const isAdmin = user && user.role === 'ADMIN';
+  const isWriter =
+    source &&
+    source.accessControls.filter(
+      acl => acl.role === 'WRITER' && acl.user.id === user.id
+    ).length > 0;
```

Comment:
would it be useful to do that in the backend ? we would update the `structureDisplay` resolver to only return attributes with inpoutGroups in case the user is a READER. 
I'm not sure it's useful though

Reply:
After discussion, we stated that this won't be done at the server level